scripts/core/program.yml

```diff
@@ -10,6 +10,65 @@ type: header
 desc: "Intel $OneApi Level-Zero Runtime APIs for Program"
 ordinal: "2"
 --- #--------------------------------------------------------------------------
+type: enum
+desc: "Program metadata property type."
+name: $x_program_metadata_type_t
+etors:
+    - name: UINT32
+      desc: "type is a 32-bit integer."
+    - name: UINT64
+      desc: "type is a 32-bit integer."
+    - name: BYTE_ARRAY
+      desc: "type is a byte array."
+    - name: STRING
+      desc: "type is a null-terminated string."
+--- #--------------------------------------------------------------------------
+type: union
+desc: "Program metadata value union."
+class: $xProgram
+name: $x_program_metadata_value_t
+members:
+    - type: char*
+      name: pString
+      desc: "[in] pointer to null-terminated string data."
+    - type: void*
+      name: pData
+      desc: "[in] pointer to binary data."
+    - type: uint64_t
```

Comment:
Will a `uint32_t` be store here?

i.e. 
```cpp
uint32_t my_data = 42;
ur_program_metadata_value_t md_value{.data = static_cast<uint64_t>(my_data)};
ur_program_metadata_t md{ "my_metadata", UR_PROGRAM_METADATA_TYPE_UINT32, 0, md_value};

```

Reply:
Correct.

Reply:
Given that this is a union, are we gaining anything by not having a dedicated `uint32_t` field? Fewer fields?

Reply:
That's a good point, the original PI interface was not a union. I'll make this a separate member.